misk-crypto/src/main/kotlin/misk/crypto/pgp/PgpEncrypterProvider.kt

```diff
@@ -0,0 +1,25 @@
+package misk.crypto.pgp
+
+import com.google.inject.Provider
+import misk.crypto.Key
+import misk.crypto.KeyReader
+import org.bouncycastle.openpgp.PGPPublicKeyRingCollection
+import org.bouncycastle.openpgp.PGPUtil
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator
+import java.io.BufferedInputStream
+
+internal class PgpEncrypterProvider(
+  private val key: Key,
+  private val isReal: Boolean
+) : Provider<PgpEncrypter>, KeyReader() {
+  override fun get(): PgpEncrypter {
+    val keyIn = BufferedInputStream(key.encrypted_key.value.byteInputStream())
+    val pgpPub =
+        PGPPublicKeyRingCollection(PGPUtil.getDecoderStream(keyIn), JcaKeyFingerprintCalculator())
+    val publicKey = pgpPub.keyRings.next().publicKey
+    return when (isReal) {
```

Comment:
This should be taken care of in the testing module, the provider shouldn't have a `isReal` variable

Reply:
well, whether or not i make a `RealPgpEncrypterProvider.kt` and `FakePgpEncrypterProvider.kt` or use a boolean inside a single provider seems kind of irrelevant to me?

can you explain more as to what you're envisioning here?